### PR TITLE
libc: optimize abs() function

### PR DESCRIPTION
**Pull request by ravisankarg**
Current version uses branching, hence wring branch predictions could be possible for some inputs. So, proposing a version without branching!

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: lib/libc/stdlib/lib_abs.c [hunk 1]
```diff
@@ -63,8 +63,9 @@
 
 int abs(int j)
 {
-	if (j < 0) {
-		j = -j;
-	}
-	return j;
+	int mask;
+
+	mask = (j >> ((sizeof(j)*8) - 1));
```
Comment (jeongchanKim) [inline lib/libc/stdlib/lib_abs.c]: please add space between operator and operand

**Comment by sunghan-chang** [comment]
@ravisankarg I don't understand exactly. Do you mean increasing number of calculations is better than branching (if)?

File: lib/libc/stdlib/lib_abs.c (hunk 1 above)
Comment (ravisankarg) [reply lib/libc/stdlib/lib_abs.c]: thanks for the review!. I guess, I will wait for #978 to merged before amending this with your comments.

**Comment by ravisankarg** [comment]
@sunghan-chang , yes executing couple of more instructions is better than pipeline flush.
& this can be at compile time. "((sizeof(j)*8) - 1)".

**Comment by junmin-kim** [comment]
@ravisankarg If user want to optimized code, user will use optimization option.
In that case, both function don't flush pipeline flush.
When I use ARM gcc 4.6.4 with -O3 option, both of abs function use 3 instruction.
I don't know which function is faster in optimized code. Are you sure that your code works faster?
```
abs_new(int):
  add r3, r0, r0, asr #31
  eor r0, r3, r0, asr #31
  bx lr
abs_before(int):
  cmp r0, #0
  rsblt r0, r0, #0
  bx lr
```

**Comment by ravisankarg** [comment]
@junmin-kim , I guess with -O3 both looks similar for me.

**Comment by ravisankarg** [comment]
thanks all for the reviews!.

**Pull request closed without merging**